Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2507 @@
+            let maxMajorTickInterval = Int(ceil(Double(tickCount / 2)))
+            
+            // Calculate the number of ticks between each major tick and the index of the first major tick
+            for i in majorTickInterval..<maxMajorTickInterval  {
+                let prospectiveInterval = i
+                var allowsEqualNumberOfTicksOnEnds = false
+                
+                // Check that the prospective interval between major ticks results in
+                // an equal number of minor ticks on both ends.
+                for m in stride(from: prospectiveInterval, to: tickCount, by: prospectiveInterval) {
+                    let totalNumberOfTicksOnEnds = tickCount - m + 1
+                    
+                    // If the total number of minor ticks on both ends (i.e. before and after the
+                    // first and last major ticks) is less than the major tick interval being tested, then we've
+                    // found the number of minor ticks that would be on the ends if we use this major tick interval.
+                    // If that total is divisible by two, then the major tick interval under test allows for an
+                    // equal number of minor ticks on the ends.
+                    if (totalNumberOfTicksOnEnds / 2 < prospectiveInterval && totalNumberOfTicksOnEnds % 2 == 0) {
+                        allowsEqualNumberOfTicksOnEnds = true
+                        break
+                    }
+                }
+                
+                // Only consider intervals that leave an equal number of ticks on the ends
+                if !allowsEqualNumberOfTicksOnEnds {
+                    continue
+                }
+                
+                // Calculate the tick index of the first major tick if we were to use the prospective interval.
+                // The index is calculated such that there will be an equal number of minor ticks before and
+                // after the first and last major tick mark.
+                firstMajorTickIndex = Int(trunc((Double(tickCount - 1).truncatingRemainder(dividingBy: Double(prospectiveInterval))) / 2.0))
+                
+                // With the given positioning of major tick marks, check whether their labels will overlap.
+                for j in stride(from: firstMajorTickIndex, to: tickCount - prospectiveInterval, by: i) {
+                    //
+                    // Get the current tick and it's label
+                    let currentTick = tms[j]
+                    var currentTickLabelFrame: CGRect? = nil
+                    if let currentTickDate = currentTick.value, let currentTickXPosition = currentTick.originX {
+                        let currentTickLabelString = timeStepIntervalDateFormatter.string(from: currentTickDate)
+                        let currentTickLabel = tickMarkLabel(with: currentTickLabelString, originX: currentTickXPosition)
+       
```

Comment:
Is it possible to break out of a `forEach` loop?  Once you find a time aware layer, you don't need to continue...

Reply:
... or maybe just `return true` instead of setting `isTimeAware = true`

Reply:
fixed